### PR TITLE
redesign transactional emails

### LINKED ISSUE #8218: Email Templates
https://github.com/gitcoinco/creative/issues/312

Screen Shot 2021-01-08 at 1 45 35 AM

### LINKED ISSUE #8355: Apply Email Templates
### Who

Gitcoin users who receive our new transactional marketing emails

### What 

should receive them based on our marketing funnels and current email architecture

### Why

so we can better understand their journey, and make sure they receive updated designs on the updates they opt into.

### Description

- [ ] apply newly designed email templates to our current email architecture setup / all highlights in yellow need to be updated to the new design style: https://docs.google.com/spreadsheets/d/1d4RToJQ9712rwI32ehKBLDT5cNn926tGT8M7QscFKNk/edit#gid=0

### PR DESCRIPTION
**Pull request by chibie**
##### Description

workflow checklist:

- [x] create test folder
- [x] redesign header + footer in base template for emails
- [x] complete the checklist below
- [x] revisit on responsivity
- [x] add UTMs to links
- [x] add product logos to email types - hackathons, bounties, grants, kudos, quests
- [x] replace svgs with pngs
- [x] implement F pattern on all emails
- [x] fix broken trans tags in emails

email checklist:

- [x] /new_bounty_daily
- [x] /new_bounty
- [x] /tribe_hackathon_prizes
- [x] /featured_funded_bounty
- [x] /new_work_submission
- [x] /new_bounty_rejection
- [x] /new_bounty_acceptance
- [x] /bounty_expire_warning
- [x] /bounty_feedback [NOT APPLICABLE - txt only]
- [x] /funder_stale [NOT APPLICABLE - txt only]
- [x] /start_work_expire_warning
- [x] /start_work_expired
- [x] /share_bounty
- [x] /start_work_approved
- [x] /start_work_rejected
- [x] /start_work_new_applicant
- [x] /start_work_applicant_about_to_expire
- [x] /start_work_applicant_expired
- [x] /funder_payout_reminder
- [x] /no_applicant_reminder
- [x] /faucet
- [x] /faucet_rejected
- [x] /request_amount_email
- [x] /comment
- [x] /mention
- [x] /wallpost
- [x] /grant_cancellation
- [x] /new_grant
- [x] /new_grant_approved
- [x] /new_supporter
- [x] /thank_you_for_supporting
- [x] /successful_contribution
- [x] /pending_contributions
- [x] /grant_update
- [x] /grant_recontribute
- [x] /grant_txn_failed
- [x] /match_distribution
- [x] /new_kudos
- [x] /kudos_mint
- [x] /kudos_mkt
- [x] /new_tip

##### Refers/Fixes

closes #8218 
closes #8355 

##### Testing

https://www.loom.com/share/4681d22aa6d344979290076366598e9d

https://www.loom.com/share/3d66f3f1450a47049ff8d6be9d09958a

**nb:** observe the hitherto broken emails now work

### CODE REVIEW

File: app/retail/utils.py [hunk 1]
```diff
@@ -792,3 +792,15 @@ def articles():
             'alt': 'open kudos'
         }
     ]
+
+
+def build_utm_tracking(email_name):
+    from urllib.parse import urlencode
```
Comment (thelostone-mc) [inline app/retail/utils.py:798]: thoughts on moving this import towards the top ?

**Comment by chibie** [comment]
screen recording update featuring, responsive design and F patterned alignment + font hierarchy fixes:

https://www.loom.com/share/3d66f3f1450a47049ff8d6be9d09958a

@PixelantDesign

**Review by octavioamu — APPROVED** [review]

**Review by gdixon — APPROVED** [review]
All looks good to me!! Good job Chibie!! 😄

File: app/retail/templates/emails/shared_template_head.html [hunk 2]
```diff
@@ -14,10 +14,6 @@
   You should have received a copy of the GNU Affero General Public License
   along with this program. If not, see <http://www.gnu.org/licenses/>.
 {% endcomment %}
-{% load i18n static compress %}
-{% compress css inline email_head %}
-  <link rel="stylesheet" type="text/x-scss" href={% static "v2/scss/gitcoin.scss" %} />
-  <link rel="stylesheet" type="text/x-scss" href={% static "v2/scss/town_square.scss" %} />
-  <link rel="stylesheet" type="text/x-scss" href={% static "v2/scss/activity_stream.scss" %} />
-{% endcompress %}
```
Comment (gdixon) [inline app/retail/templates/emails/shared_template_head.html:22]: Hey @chibie, I've just tried this change out on the stable branch, and we are using these styles for a few of the emails. Could you have a look at #8553 and reintroduce the inline css as it's done there 👍 😄 Thank you!!

**Review by gdixon — CHANGES_REQUESTED** [review]
Could you double check that the changes you've made mean that we don't need those 3 css documents. I checked `new_bounty_daily` and we are using the .css to style the activity feed.

File: app/retail/templates/emails/shared_template_head.html (hunk 2 above)
Comment (chibie) [reply app/retail/templates/emails/shared_template_head.html:22]: @gdixon

`new_bounty_daily` will be purged in favour of `new_bounty` which has the current requirement for the **daily digest** email (activities, quests and stuff were taken out)

can you checkout staging to this pr branch so we regress there?

**Comment by frankchen07** [comment]
sorry I said EOD on Friday but failed. here's my review:

```
stage.gitcoin.co/_administration/email/new_bounty_daily
    left indent issue for the content else lgtm
stage.gitcoin.co/_administration/email/new_bounty
    left indent issue for the content else lgtm
stage.gitcoin.co/_administration/email/tribe_hackathon_prizes
    error nothing shows up
stage.gitcoin.co/_administration/email/featured_funded_bounty
    left indented, is this intended? else lgtm
stage.gitcoin.co/_administration/email/new_work_submission
    left indented, is this intended? else lgtm
stage.gitcoin.co/_administration/email/new_bounty_rejection
    left indented, is this intended? else lgtm
stage.gitcoin.co/_administration/email/new_bounty_acceptance
    left indented, is this intended? else lgtm
stage.gitcoin.co/_administration/email/bounty_expire_warning
    left indented, is this intended?
    capitalize github
    replace slack with discord & correct link?
stage.gitcoin.co/_administration/email/bounty_feedback [NOT APPLICABLE - txt only]
    error nothing shows up
stage.gitcoin.co/_administration/email/funder_stale [NOT APPLICABLE - txt only]
    lgtm
stage.gitcoin.co/_administration/email/start_work_expire_warning
    left indented, is this intended?
    replace slack with discord & correct link?
stage.gitcoin.co/_administration/email/start_work_expired
    left indented, is this intended?
    replace slack with discord & correct link?
stage.gitcoin.co/_administration/email/share_bounty
    broken image but because it's testing? else lgtm
stage.gitcoin.co/_administration/email/start_work_approved
    left indented, is this intended? else lgtm
stage.gitcoin.co/_administration/email/start_work_rejected
    left indented, is this intended? else lgtm
stage.gitcoin.co/_administration/email/start_work_new_applicant
    left indented, is this intended? else lgtm
stage.gitcoin.co/_administration/email/start_work_applicant_about_to_expire
    left indented, is this intended? else lgtm
stage.gitcoin.co/_administration/email/start_work_applicant_expired
    left indented, is this intended? else lgtm
stage.gitcoin.co/_administration/email/funder_payout_reminder
    error nothing shows up
stage.gitcoin.co/_administration/email/no_applicant_reminder
    left indented, is this intended?
    at some point we'll want to revisit this, user directory is dead
stage.gitcoin.co/_administration/email/faucet
    error nothing shows up
stage.gitcoin.co/_administration/email/faucet_rejected
    error nothing shows up
stage.gitcoin.co/_administration/email/request_amount_email
    error nothing shows up
stage.gitcoin.co/_administration/email/comment
    left indented, is this intended? else lgtm
stage.gitcoin.co/_administration/email/mention
    left indented, is this intended? else lgtm
stage.gitcoin.co/_administration/email/wallpost
    left indented, is this intended? else lgtm
stage.gitcoin.co/_administration/email/grant_cancellation
    left indented, is this intended? else LGTM
stage.gitcoin.co/_administration/email/new_grant
    left indented, is this intended? else lgtm
stage.gitcoin.co/_administration/email/new_grant_approved
    left indented, is this intended?
    same as /new_grant, intended? else lgtm
stage.gitcoin.co/_administration/email/new_supporter
    left indented, is this intended? else lgtm
stage.gitcoin.co/_administration/email/thank_you_for_supporting
    left indented, is this intended? else lgtm
stage.gitcoin.co/_administration/email/successful_contribution
    left indented, is this intended? else lgtm
stage.gitcoin.co/_administration/email/pending_contributions
    left indented, is this intended? else lgtm
stage.gitcoin.co/_administration/email/grant_update
    left indented, is this intended? 
    unsubscribes are quite large, else lgtm
stage.gitcoin.co/_administration/email/grant_recontribute
    error nothing shows up
stage.gitcoin.co/_administration/email/grant_txn_failed
    error nothing shows up
stage.gitcoin.co/_administration/email/match_distribution
    looks incomplete
stage.gitcoin.co/_administration/email/new_kudos
    no image link, but I think it's b/c it's testing, lgtm
stage.gitcoin.co/_administration/email/kudos_mint
    lgtm
stage.gitcoin.co/_administration/email/kudos_mkt
    left indented, is this intended? else lgtm
stage.gitcoin.co/_administration/email/new_tip
    left indented, is this intended? else lgtm
```

**Review by thelostone-mc — APPROVED** [review]

**Comment by PixelantDesign** [comment]
Looks good overall! Daily digest looks really good! Is it possible to have all of the templates follow the way the daily digest changes between web and responsive views as well as treatment of headers?

On this email
https://stage.gitcoin.co/_administration/email/featured_funded_bounty
Could we center the gitcoin bot illustration? See daily digest email 
Could we remove this text: (& Being Sent To Gitcoin's Worldwide Community Of Developers)

**Comment by chibie** [comment]
@PixelantDesign 

> On this email
> https://stage.gitcoin.co/_administration/email/featured_funded_bounty
> Could we center the gitcoin bot illustration? See daily digest email
> Could we remove this text: (& Being Sent To Gitcoin's Worldwide Community Of Developers)

done.

> Is it possible to have all of the templates follow the way the daily digest changes between web and responsive views as well as treatment of headers?

can you explain a bit more?

**Pull request merged**